src/algebraic_geometry/pullbacks.lean

```diff
@@ -0,0 +1,330 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebraic_geometry.gluing
+import category_theory.limits.opposites
+import algebraic_geometry.Gamma_Spec_adjunction
+
+/-!
+# Fibred products of schemes
+
+In this file we construct the fibred product of schemes via gluing.
+We roughly follow [har77] Theorem 3.3.
+
+In particular, the main construction is to show that for an open cover `{ Uᵢ }` of `X`, if there
+exist fibred products `Uᵢ ×[Z] Y` for each `i`, then there exists a fibred product `X ×[Z] Y`.
+
+Then, for constructing the fibred product for arbitrary schemes `X, Y, Z`, we can use the
+construction to reduce to the case where `X, Y, Z` are all affine, where the fibred products is
+constructed via tensor products.
+
+-/
+universes v u
+noncomputable theory
+
+open category_theory category_theory.limits algebraic_geometry
+namespace algebraic_geometry.Scheme
+
+variables {C : Type u} [category.{v} C]
+
+variables {X Y Z : Scheme.{u}} (𝒰 : open_cover.{u} X) (f : X ⟶ Z) (g : Y ⟶ Z)
+variables [∀ x, has_pullback (𝒰.map x ≫ f) g]
+
+/-- The intersection of `Uᵢ ×[Z] Y` and `Uⱼ ×[Z] Y` is given by (Uᵢ ×[Z] Y) ×[X] Uⱼ -/
+def V (x y : 𝒰.J) : Scheme :=
+pullback ((pullback.fst : pullback ((𝒰.map x) ≫ f) g ⟶ _) ≫ (𝒰.map x)) (𝒰.map y)
+
+/-- The canonical transition map `(Uᵢ ×[Z] Y) ×[X] Uⱼ ⟶ (Uⱼ ×[Z] Y) ×[X] Uᵢ` given by the fact
+that pullbacks are associative and symmetric. -/
+def t (x y : 𝒰.J) : V 𝒰 f g x y ⟶ V 𝒰 f g y x :=
```

Comment:
Idem dito, just to match the docstrings, and the conventional notation in informal maths.